s3-bucket-codepipeline.tf

```diff
@@ -1,18 +1,33 @@
-#this bucket is used to store config files, etc. which are used for processing.
+# This bucket is used to store config files, etc. which are used for processing
 resource "aws_s3_bucket" "pipeline_resources_bucket" {
   bucket = var.name
 
-  versioning {
-    enabled = true
-  }
-
   lifecycle {
     prevent_destroy = false
   }
 
-  lifecycle_rule {
-    id      = "artifacts"
-    enabled = true
+#    logging {
+#      target_bucket = var.s3_log_target_bucket
+#      target_prefix = "s3/${var.name}/"
+#    }
+
+  tags = merge(var.input_tags, {})
+}
+
+# Versioning enabled in Pipeline bucket
+resource "aws_s3_bucket_versioning" "pipeline_resources_bucket" {
+  bucket = aws_s3_bucket.pipeline_resources_bucket.id
+  versioning_configuration {
+    status = "Enabled"
+  }
+}
+
+# Lifecycle options for Pipeline bucket
+resource "aws_s3_bucket_lifecycle_configuration" "pipeline_resources_bucket" {
+  bucket = aws_s3_bucket.pipeline_resources_bucket.id
```

Comment:
This probably needs to reference the `bucket` attribute: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/s3_bucket_lifecycle_configuration#bucket